lib/dao.js

```diff
@@ -856,8 +856,9 @@ DataAccessObject.findById = function findById(id, filter, options, cb) {
   assert(typeof options === 'object', 'The options argument must be an object');
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
-  if (!this.definition.hasIdDefined()) {
-    process.nextTick(cb(getIdDefinitionError()));
+  if (!this.definition.hasPrimaryKey()) {
+    process.nextTick(cb(new PrimaryKeyMissingError(this.modelName)));
```

Comment:
This is not going to work. `process.nextTick()` takes a function. The code should look like the following:
```js

   // Define a helper function
   function isPKMissing(modelClass, cb) {
     var hasPK = modelClass.definition.hasPrimaryKey();
     if (hasPK) return false;
     process.nextTick(function() {
       cb(new PrimaryKeyMissingError(modelClass.modelName);
     }
     return true;
 }

// Use the helper function
if (isPKMissing(this, cb)) return cb.promise;
```